### PR TITLE
Replace almost all uses of 'syscall' with 'unix' pkg.

### LINKED ISSUE #10116: Remove references to `syscall` package
The [`syscall`](https://golang.org/pkg/syscall/) package is deprecated since Go 1.4:
> Deprecated: this package is locked down. Callers should use the corresponding package in the golang.org/x/sys repository instead. That is also where updates required by new systems or versions should be applied. See https://golang.org/s/go1.4-syscall for more information.

References to the `syscall` package in the codebase should be cleaned up as it is, in effect, not updated [_"not even to keep pace with changes in operating systems it references"_](https://golang.org/s/go1.4-syscall).

### PR DESCRIPTION
**Pull request by Ropes**
`golang.org/x/sys/unix` package is replacement for `syscall` which has
been deprecated since Go 1.4.

> Deprecated: this package is locked down. Callers should use the corresponding package in the golang.org/x/sys repository instead. That is also where updates required by new systems or versions should be applied. See https://golang.org/s/go1.4-syscall for more information.

While the changes do touch a lot of files, all of the modifications are
simply replacing imports of the `syscall` package, with `x/sys/unix`
implementations. No functionality changes, so no new tests added.

There is one case of `syscall` remaining in the cilium dnsproxy. There
is currently [no good replacement](https://github.com/cilium/cilium/issues/10116#issuecomment-584838627) listening for UDP traffic.

Fixes: https://github.com/cilium/cilium/issues/10116

Signed-off-by: Joshua Roppo <joshroppo@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/10158)

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by Ropes** [comment]
Alright, I've been able to verify the test failure locally. Also checked the prior commit to make sure that there wasn't some upstream instability which sneaked through. 

Very curious what has actually been uncovered here. I'll start investigating `syscall` vs `unix` package constants, perhaps there is a difference between values with the same name? Seems very peculiar if that were the case, but the change surface is rather small.

**Comment by joestringer** [comment]
@Ropes As far as I can tell, there's only one failure which looks like the known CI flake #9869 :

https://jenkins.cilium.io/job/Cilium-PR-Ginkgo-Tests-Validated/17085/testReport/junit/Suite-k8s-1/17/K8sPolicyTest_Basic_Test_TLS_policy/

So if the rest of the build (including the full testsuite with the k8s-1.11+net-next+kubeproxy-free) passes then this should be good enough from a testing perspective. Just need reviews from the codeowners.

**Review by brb — APPROVED** [review]
LGTM, thanks!

There will be a potential conflict once https://github.com/cilium/cilium/pull/10117 gets merged.

**Review by rolinh — APPROVED** [review]
LGTM. And there might be yet another conflict with #10058 once it's merged.

**Comment by Ropes** [comment]
Since there aren't really any logic changes in this PR, it'll be easier to refactor post #10058 & #10117 get merged. When they merge I'll have a second pass and get it ready again. #10117 is similar to what this changeset.. just didn't pick up using `unix` package.

**Comment by Ropes** [comment]
Just force pushed a fix for a code conflict with master.

File: pkg/maps/policymap/policymap_privileged_test.go
Comment (rolinh) [inline pkg/maps/policymap/policymap_privileged_test.go]: Why was this assertion removed?

File: api/v1/health/server/server.go
Comment (tklauser) [inline api/v1/health/server/server.go]: This file is autogenerated, so this change will be reverted the next time it's generated. You'd probably need to change `go-swagger` to generate `x/sys/unix` instead of `syscall`.

File: cilium-health/responder/main.go
Comment (tklauser) [inline cilium-health/responder/main.go]: Move this import below to the non-cilium imports.

File: cilium/cmd/cleanup.go
Comment (tklauser) [inline cilium/cmd/cleanup.go]: Move this import below to the non-cilium imports.

File: daemon/cleanup.go
Comment (tklauser) [inline daemon/cleanup.go:26]: Add an empty line before this import to separate cilium imports from non-cilium imports.

File: operator/main.go
Comment (tklauser) [inline operator/main.go]: Move this import below to the non-cilium imports.

File: pkg/bpf/bpffs_linux.go
Comment (tklauser) [inline pkg/bpf/bpffs_linux.go:29]: Add an empty line before this import.

File: pkg/cgroups/cgroups.go
Comment (tklauser) [inline pkg/cgroups/cgroups.go:27]: Add an empty line before this import.

File: pkg/datapath/prefilter/prefilter.go
Comment (tklauser) [inline pkg/datapath/prefilter/prefilter.go:31]: Add an empty line before this import.

File: pkg/monitor/agent/listener/listener.go
Comment (tklauser) [inline pkg/monitor/agent/listener/listener.go:23]: Add an empty line before this import.

File: pkg/probe/probe.go
Comment (tklauser) [inline pkg/probe/probe.go:23]: Add an empty line before this import.

**Review by tklauser — CHANGES_REQUESTED** [review]

File: api/v1/health/server/server.go
Comment (Ropes) [reply api/v1/health/server/server.go]: Good catch, resolving this has opened up a bit of a yak shaving thread.

Effectively cilium:master has stale swagger generated sever code, and which is hoping to be rectified by: https://github.com/cilium/cilium/pull/10336

Comment (tklauser) [reply api/v1/health/server/server.go]: Thanks for taking care of it!

File: pkg/maps/policymap/policymap_privileged_test.go
Comment (Ropes) [reply pkg/maps/policymap/policymap_privileged_test.go]: I'm uncertain.. My guess is that the change manifests here after rebasing master. 

Currently sorting out issues with swagger: https://github.com/cilium/cilium/pull/10336 Once that's done I'll do another sweep on these changes.

File: pkg/monitor/agent/listener/listener.go
Comment (Ropes) [reply pkg/monitor/agent/listener/listener.go:23]: I'll clean all these imports up after we get swagger and master fixed. https://github.com/cilium/cilium/pull/10336

File: pkg/maps/policymap/policymap_privileged_test.go
Comment (tklauser) [inline pkg/maps/policymap/policymap_privileged_test.go]: Why was this line removed? Merge gone wrong maybe?

Comment (jrajahalme) [reply pkg/maps/policymap/policymap_privileged_test.go]: Variable `log` is maybe shadowing the newly introduced system package "log". It seems that elsewhere we import `"log"` as `stdlog "log"` to avoid this (e.g., [pkg/policy/rule_test.go#L21](https://github.com/cilium/cilium/blob/master/pkg/policy/rule_test.go#L21)).

Changes in this file seem unrelated to this PR, however.

File: api/v1/health/server/server.go
Comment (Ropes) [reply api/v1/health/server/server.go]: I've removed this change to the generated file. `/api/v1/server.gotmpl` has been updated to use the `unix` package. However there are breaking changes when running `swagger` at the moment. I [outlined](https://github.com/cilium/cilium/pull/10336#issuecomment-591107088) the issues in #10336, once those are resolved, regenerating the APIs should pick up the unix package changes.

File: pkg/maps/policymap/policymap_privileged_test.go
Comment (Ropes) [reply pkg/maps/policymap/policymap_privileged_test.go]: I'm not certain why it was modified, only thing I can think is that VSCode might have done it in a `goimports` or `govet`. I've removed the change so there's no difference from master.

File: pkg/envoy/accesslog_server.go
Comment (joestringer) [inline pkg/envoy/accesslog_server.go]: Why is this line changing if there's no corresponding change in the rest of the file?

File: pkg/maps/policymap/policymap_privileged_test.go
Comment (joestringer) [inline pkg/maps/policymap/policymap_privileged_test.go]: Does this line need to be here?

File: proxylib/test/accesslog_server.go
Comment (joestringer) [inline proxylib/test/accesslog_server.go]: This also seems out-of-place if there's no corresponding line change for users of this import.

**Review by joestringer — CHANGES_REQUESTED** [review]
Just a few remaining unexpected changes, otherwise LGTM.

File: proxylib/test/accesslog_server.go
Comment (Ropes) [reply proxylib/test/accesslog_server.go]: I need to smash vscode in the head.. I am unsure why that changed... will fix.

File: pkg/envoy/accesslog_server.go
Comment (Ropes) [reply pkg/envoy/accesslog_server.go]: I believe the problem was: `'goreturns' will add missing imports and remove unused imports.` 

Disabled.... vscode would make the change before saving and closing the file.. :man_facepalming:.

Comment (joestringer) [reply pkg/envoy/accesslog_server.go]: Are there many vscode-specific configurations you need to apply for developing on Cilium? Just wondering if it's worth sharing the vscode configs in the Cilium repo or as documentation or even just as a gist somewhere so if someone searches "vscode cilium" they'll find the instructions?

**Comment by joestringer** [comment]
test-me-please

**Review by joestringer — APPROVED** [review]

**Review by tklauser — APPROVED** [review]
LGTM, thanks.

**Pull request merged**
